queries/html/rainbow-delimiters.scm

```diff
@@ -1,19 +1,47 @@
-;;; A pair of opening and closing tag with any content in-between. Excludes
-;;; self-closing tags or opening tags without closing tag.
+;;; A pair of delimiter and closing tag with any content in-between. Excludes
+;;; self-closing tags or delimiter tags without closing tag.
 
 (element
-  (start_tag (tag_name) @opening)
-  ; (element (self_closing_tag) @intermediate)*
-  (end_tag (tag_name) @closing)) @container
+  (start_tag
+    "<" @delimiter
+    (tag_name) @delimiter
+    ">" @delimiter)
+  (end_tag
+    "</" @delimiter
+    (tag_name) @delimiter
+    ">" @delimiter @sentinel)) @container
 
 (element
-  (self_closing_tag (tag_name) @opening)) @container
+  (self_closing_tag
+    "<" @delimiter
+    (tag_name) @delimiter
+    "/>" @delimiter @sentinel)) @container
+
+;; The slash in a self-closing tag is optional
+;; This pattern does not actually work, it conflicts with first pattern.
+; (element
+;   (start_tag
+;     "<" @delimiter
+;     (tag_name) @delimiter
+;     ">" @delimiter @sentinel)) @container
```

Comment:
If we really want the self-closing tags without writing the `/`, then we can do something like this:
```suggestion
(element
  (start_tag
    "<" @delimiter
    (tag_name) @delimiter @ignore
    ">" @delimiter @sentinel)
  (#any-of? @ignore
   "area"
   "base"
   "br"
   "col"
   "embed"
   "hr"
   "img"
   "input"
   "link"
   "meta"
   "param"
   "source"
   "track"
   "wbr")
) @container
```

Reply:
I like this. It will only work for select tag names, but it's a reasonably-sized list and it is easy to extend.